### PR TITLE
BUG-101035 underscore '_' is enabled for hostgroup name in blueprint

### PR DESCRIPTION
**Pull request by attilapalfi92**
Closes #BUG-101035

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/converter/BlueprintRequestToBlueprintConverter.java [hunk 1]
```diff
@@ -148,14 +148,18 @@ private void validateHostGroups(JsonNode root) {
             JsonNode hostGroupName = hostGroup.path("name");
             if (isTextPropertyMissing(hostGroupName)) {
                 throw new BadRequestException("Validation error: one of the 'host_groups' has no name.");
-            } else if (!StringUtils.isAlphanumeric(hostGroupName.asText())) {
+            } else if (isInvalidHostgroupName(hostGroupName)) {
                 throw new BadRequestException(String.format("Validation error: '%s' is not a valid host group name. Host group name must be alphanumeric.",
```
Comment (mhmxs) [inline core/src/main/java/com/sequenceiq/cloudbreak/converter/BlueprintRequestToBlueprintConverter.java]: The error message hasn't extended with new criteria

File: core/src/main/java/com/sequenceiq/cloudbreak/converter/BlueprintRequestToBlueprintConverter.java [hunk 2]
```diff
@@ -148,14 +148,18 @@ private void validateHostGroups(JsonNode root) {
...
                         hostGroupName.asText()));
             }
             validateComponentsInHostgroup(hostGroup, hostGroupName.asText());
         }
     }
 
+    private boolean isInvalidHostgroupName(JsonNode hostGroupName) {
+        return !StringUtils.isAlphanumeric(hostGroupName.asText().replaceAll("-", ""));
```
Comment (mhmxs) [inline core/src/main/java/com/sequenceiq/cloudbreak/converter/BlueprintRequestToBlueprintConverter.java]: Not so flexible solution, would be better an extendable solution.

**Comment by keyki** [comment]
Dash is not allowed, underscore is

File: core/src/main/java/com/sequenceiq/cloudbreak/converter/BlueprintRequestToBlueprintConverter.java (hunk 2 above)
Comment (attilapalfi92) [reply core/src/main/java/com/sequenceiq/cloudbreak/converter/BlueprintRequestToBlueprintConverter.java]: I'll move it to blueprint utils

**Pull request merged**